### DISCUSSION TITLE
It seems that the build fails also on Ubuntu and Windows.

### DISCUSSION THREAD

**Discussion by SekouDiaoNlp** [juancarlospaco/faster-than-csv#27] [General]
last workflow to run succesfully on Ubuntu and windows was [Update faster_than_csv.nim.cfg Build #154](https://github.com/juancarlospaco/faster-than-csv/actions/runs/829483093) on the 10th of may.

I reset the the build.yml to [the last known build to succed on Ubuntu and Windows](https://github.com/juancarlospaco/faster-than-csv/commit/eab234f04b0e85875ca749a8ea8526cad76e5f99) and still the same issue.

I don't think this issue is only related to MacOS.

I will investigate more deeply this week what changes were made to `nim >=1.4` to try to either fix nimporter or isolate the issue and find a fix.

**Comment by juancarlospaco** [comment]
In the past I tried shipping the C sources, works Ok with Nim, 
but Python/PIP sucks for that, it wont allow to customize anything like C compiler options etc,
they also change how C compilers are called in minor releases without notice, and is not documented,
in a last resort option that can be restored, it should be in the Git history.

I also tried to make a PIP that works, but I never finished it sadly, 
Python wont document how PIP works, so is mostly done by reverse engineering manually
https://github.com/juancarlospaco/plz#plz

I seen other PIP alternatives, but they just import PIP, they wont work without PIP, thats kinda pointless, 
mine works even without PIP installed.

Another option is to remove Nimporter, depend on `choosenim_install`, 
make `setup.py` call the Nim compiler using `subprocess.run("nim c " + file)` to generate a `.so`.

Too bad PIP is not as good as Cargo or Nimble.

**Reply by SekouDiaoNlp** [reply]
> In the past I tried shipping the C sources, works Ok with Nim,
> but Python/PIP sucks for that, it wont allow to customize anything like C compiler options etc,
> they also change how C compilers are called in minor releases without notice, and is not documented,
> in a last resort option that can be restored, it should be in the Git history.
> 

Yes at worse we can resort to that.

> I also tried to make a PIP that works, but I never finished it sadly,
> Python wont document how PIP works, so is mostly done by reverse engineering manually
> https://github.com/juancarlospaco/plz#plz
> 
> I seen other PIP alternatives, but they just import PIP, they wont work without PIP, thats kinda pointless,
> mine works even without PIP installed.
> 

That's a cool project, too bad yo didn't finish it. How far did you advance?

> Another option is to remove Nimporter, depend on `choosenim_install`,
> make `setup.py` call the Nim compiler using `subprocess.run("nim c " + file)` to generate a `.so`.

Yes I will look into it. This my project for next week. Either fix nimporter to work or find an alternative like the one you propose.

> Too bad PIP is not as good as Cargo or Nimble.

Definitely! A more versatile tool would be most welcome imho.

**Reply by juancarlospaco** [reply]
> That's a cool project, too bad yo didn't finish it. How far did you advance?

Everything that its readme says works, or was working, it can even upload packages to Pypi so it replaces `twine` too,
it can generate "project templates" for Python, because at that time Github repo template did not exist,
needs more testing to see whats working and what is not working, and fix any bug found.

**Reply by SekouDiaoNlp** [reply]
Yes I was just having a look at the README.
Looks like a nice project to continue and promote 👍🏿

**Comment by SekouDiaoNlp** [comment]
What is weird is that the dockerfile works perfectly to build and run the benchmarks with the code as is.
It works perfectly on Ubuntu 20.04 as well as Windows 10 on both `gcc 9.3.0 , nim 1.4.2` and `gcc 10.3.0 , nim 1.5.1`.

**Reply by juancarlospaco** [reply]
The problem is on the Python side of things, not on the Nim side, Nim runs anywhere from browser to arduinos.

I fixed https://github.com/juancarlospaco/plz#plz
I can invite you if you are interested.

Let me know if you discover something about Nimporter.

**Reply by SekouDiaoNlp** [reply]
> The problem is on the Python side of things, not on the Nim side, Nim runs anywhere from browser to arduinos.
> 

Yeah that's what I figured.

> I fixed https://github.com/juancarlospaco/plz#plz
> I can invite you if you are interested.

I am definitely interested, you can add me.

> Let me know if you discover something about Nimporter.

Yes I will investigate during the week and let you know.

**Reply by juancarlospaco** [reply]
So you got the Nimporter control in the end ?.

Maybe we should create our own solution... :thinking:

**Reply by SekouDiaoNlp** [reply]
Hi, what's up? No I couldn't get nimporter to work properly.

We should implement our own fix
